utils/python/nexus_nei2016_linker.py

```diff
@@ -51,26 +55,36 @@ def get_closest_file(target_date,dates,files):
             return files[index]
         else: 
             return find_day_in_iso_week(d, dates,files)
-    elif len(files) == 7:
+    elif nfiles_per_month == 7:
         print('ONLY ONE WEEK')
         #daily files are availbale for a single month
-        index = dow.index(target_date.isoweekday())
-        return files[index]
-    elif len(files) == 4:
+        files_in_month, iwd_in_month = get_files_in_month(files, dates, target_date)
+        index = iwd_in_month.index(target_date.isoweekday())
+        return files_in_month[index]
+    elif nfiles_per_month == 4:
         print('ONLY 4 DAYS')
         # a week day and friday sat and sunday are available
+        # should return the files in the current month regardless
+        files_in_month, iwd_in_month = get_files_in_month(files, dates, target_date)
         if tiwd == 1:
-            return files[0]
+            index = iwd_in_month.index(tiwd)
+            return files_in_month[index]
         elif tiwd == 6:
-            return files[-2]
+            index = iwd_in_month.index(tiwd)
+            return files_in_month[index]
         elif tiwd == 7:
-            return files[-1]
+            index = iwd_in_month.index(tiwd)
+            return files_in_month[index]
         else:
-            return files[1]
+            index = [iwd_in_month.index(item) for item in iwd_in_month if item not in [1, 6, 7]][0]
+            return files_in_month[index]
     else:
         print('SINGLE_FILE')
         # only a single file for the entire month
-        return files[0]
+        target_month = target_date.month
+        file_month = get_month(dates)
+        index, _ = min(enumerate(file_month), key=lambda x: abs(x[1] - target_month))
```

Comment:
This line can be replaced with:
```
index = find_closest_index(file_month, target_month)
```
Function `find_closest_index` should be moved around line 40